include/mqtt/endpoint.hpp

```diff
@@ -1516,8 +1517,14 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
         variant<suback_return_code, v5::suback_reason_code> reason,
         v5::properties props = {}
     ) {
-        variant<std::vector<suback_return_code>, std::vector<v5::suback_reason_code>> params;
-        send_suback(force_move(params), packet_id, std::move(reason), std::move(props));
+        if(0 == variant_idx(reason))
+        {
+            send_suback(std::vector<suback_return_code>{ variant_get<suback_return_code>(reason) }, packet_id, force_move(props));
+        }
+        else
+        {
+            send_suback(std::vector<v5::suback_reason_code>{ variant_get<v5::suback_reason_code>(reason) }, packet_id, force_move(props));
+        }
```

Comment:
Please keep the coding style as follows:

```cpp
        if (variant_idx(reason) == 0) {
            send_suback(std::vector<suback_return_code>{ variant_get<suback_return_code>(reason) }, packet_id, force_move(props));
        }
        else {
            send_suback(std::vector<v5::suback_reason_code>{ variant_get<v5::suback_reason_code>(reason) }, packet_id, force_move(props));
        }
```

The contents is no problem.

Reply:
Fixed

Reply:
A little more update is required.

```cpp
    if(0 == variant_idx(reason)) {
```

Please insert a space between `if` and `(`.   
Please place constant to the right hand side. I know the many options about this. For example, left hand side constant can avoid typo `==` as `=`. But modern compiler outputs appropriate warnings about that. In addition the update is for consistency.

```cpp
    if (variant_idx(reason) == 0) {
```

Reply:
Thank you for explaining.